Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransforms.java

```diff
@@ -112,7 +116,7 @@ public final LogEventRequestType transformRequestToAuditMsg(T request, Assertion
         return buildLogEventRequestType(auditMsg, direction, getMessageCommunityId(assertion, target, isRequesting),
             serviceName, assertion, auditMsg.getEventIdentification().getEventID().getDisplayName(),
             auditMsg.getEventIdentification().getEventOutcomeIndicator(),
-            auditMsg.getEventIdentification().getEventDateTime(), getUserId(auditMsg.getActiveParticipant()));
+            auditMsg.getEventIdentification().getEventDateTime(), getRequestorUserId(auditMsg,assertion));
```

Comment:
I think you can check before the return statement about null criteria for auditMsg, as auditMsg is used for buildLogEventRequestType method's parameters